### PR TITLE
Use slice patterns

### PR DESCRIPTION
**Pull request by Kixunil**
Some code looks better with slice patterns. This changes `bip32` to use them.

### CODE REVIEW

**Review by apoelstra — APPROVED** [review]
ACK 873501a85f1e579312c167fff9fba95a9a6130ce

File: bitcoin/src/bip32.rs [hunk 1]
```diff
@@ -607,14 +607,11 @@ impl ExtendedPrivKey {
...
-        } else if data[0..4] == [0x04u8, 0x35, 0x83, 0x94] {
-            Network::Testnet
-        } else {
-            let mut ver = [0u8; 4];
-            ver.copy_from_slice(&data[0..4]);
-            return Err(Error::UnknownVersion(ver));
+        let network = match data {
+            [0x04u8, 0x88, 0xAD, 0xE4, ..] => Network::Bitcoin,
+            [0x04u8, 0x35, 0x83, 0x94, ..] => Network::Testnet,
+            [b0, b1, b2, b3, ..] => return Err(Error::UnknownVersion([*b0, *b1, *b2, *b3])),
```
Comment (tcharding) [inline bitcoin/src/bip32.rs:613]: Cool trick I recently learned:
```suggestion
            [b0, b1, b2, b3, ..] => Err(Error::UnknownVersion([*b0, *b1, *b2, *b3]))?,
```

**Review by tcharding — APPROVED** [review]
ACK 873501a85f1e579312c167fff9fba95a9a6130ce

File: bitcoin/src/bip32.rs (hunk 1 above)
Comment (apoelstra) [reply bitcoin/src/bip32.rs:613]: Heh, I think that's a little too clever.

Comment (Kixunil) [reply bitcoin/src/bip32.rs:613]: Even though I know that the condition inside will get elided it still feels wrong...

Comment (tcharding) [reply bitcoin/src/bip32.rs:613]: Fair points, I won't use it. I have a feeling I may have done so already - I've been using it for a week or so.

Comment (apoelstra) [reply bitcoin/src/bip32.rs:613]: Heh, it's not in the current codebase fwiw.

Comment (tcharding) [reply bitcoin/src/bip32.rs:613]: Cool, thanks.

**Pull request merged**